lib/spack/spack/bootstrap.py

```diff
@@ -975,3 +976,22 @@ def status_message(section):
         msg += '\n'
         msg = msg.format(pass_token if not missing_software else fail_token)
     return msg, missing_software
+
+
+def bootstrapping_sources(scope=None):
+    """Return the list of configured sources of software for bootstrapping Spack
+
+    Args:
+        scope (str or None): if a valid configuration scope is given, return the
+            list only from that scope
+    """
+    source_configs = spack.config.get('bootstrap:sources', default=None, scope=scope)
+    source_configs = source_configs or []
+    list_of_sources = []
+    for entry in source_configs:
+        current = copy.copy(entry)
+        metadata_file = spack.util.path.canonicalize_path(entry['metadata'])
+        with open(metadata_file) as f:
+            current.update(ruamel.yaml.load(f))
```

Comment:
We should probably use spack.util.yaml here for consistency with other config values